mlx_lm/server.py

```diff
@@ -534,8 +534,12 @@ def handle_completion(
                 sorted_indices = mx.argpartition(-logprobs, kth=self.logprobs - 1)
                 top_indices = sorted_indices[: self.logprobs]
                 top_logprobs = logprobs[top_indices]
-                top_token_info = zip(top_indices.tolist(), top_logprobs.tolist())
-                top_tokens.append(tuple(top_token_info))
+                top_tokens.append(
+                    {
+                        int(idx): float(prob)
+                        for idx, prob in zip(top_indices, top_logprobs)
+                    }
+                )
```

Comment:
This change is not good. We should do `tolist` and loop over Python lists rather than `mx.array`.